packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -174,10 +251,122 @@ contract FederatedAttestations is
         trimmedIssuedOns[i] = issuedOns[i];
         trimmedSigners[i] = signers[i];
       }
-      return (trimmedAccounts, trimmedIssuedOns, trimmedSigners);
+      return (countsPerIssuer, trimmedAccounts, trimmedIssuedOns, trimmedSigners);
     } else {
-      return (accounts, issuedOns, signers);
+      return (countsPerIssuer, accounts, issuedOns, signers);
+    }
+  }
+
+  /**
+   * @notice Similar to lookupAttestations but returns all attestations for
+   *   an identifier produced by a list of issuers, regardless of whether
+   *   signers have been revoked or not
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @return [0] Array of number of attestations returned per issuer
+   * @return [1 - 3] for m (== sum([0])) found attestations:
+   *         [
+   *           Array of m accounts,
+   *           Array of m issuedOns,
+   *           Array of m signers
+   *         ]; index corresponds to the same attestation
+   * @dev Adds attestation info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+  function lookupAllAttestations(bytes32 identifier, address[] calldata trustedIssuers)
+    external
+    view
+    returns (uint256[] memory, address[] memory, uint256[] memory, address[] memory)
+  {
+    uint256 totalAttestations;
+    uint256[] memory countsPerIssuer;
+
+    (totalAttestations, countsPerIssuer) = getTotalNumberOfAttestations(identifier, trustedIssuers);
+    address[] memory accounts = new address[](totalAttestations);
+    uint256[] memory issuedOns = new uint256[](totalAttestations);
+    address[] memory signers = new address[](totalAttestations);
+
+    OwnershipAttestation[] memory attestationsPerIssuer;
+    // Reset this and use as current index to get around stack-too-deep
```

Comment:
Do we get a stack too deep error if we declare another counter? (i.e. like currIndex in lookupAttestations?)

Reply:
Yes I was earlier 😬 (I can double check in case I deleted any vars around here but I believe so) It may be possible to pack totalAttestations & the counter into a struct, but not sure if it's cleaner?

Reply:
Update: still causes the stack too deep error but added a comment for reviewers on if there's a better way to deal with this